docs/community/release_notes/release_note_0.4.0.md

```diff
@@ -0,0 +1,268 @@
+- [CELEBORN-1193] ResettableSlidingWindowReservoir should reset `full` to false
+- [CELEBORN-1188][TEST] Using JUnit function instead of java assert
+- [CELEBORN-1190] Apply error prone patch and suppress some problems
+- [CELEBORN-1036][FOLLOWUP] totalInflightReqs should decrement when batchIdSet contains the batchId to avoid duplicate caller of removeBatch
+- [CELEBORN-1150] support io encryption for spark
+- [CELEBORN-1176] Server side support for Sasl Auth
+- [CELEBORN-1180] Changed the version of Sasl Auth related config to 0.5
+- [CELEBORN-1157] Add client-side support for Sasl Authentication in the transport layer
+- [CELEBORN-1164] Introduce FetchChunkFailCount metric to expose the count of fetching chunk failed in current worker
+- [CELEBORN-1162][BUG] Fix refCnt 0 Exception in FetchHandler#handleChunkFetchRequest
+- [CELEBORN-1151] Request slots when register shuffle should filter the workers excluded by application
+- [CELEBORN-1152] fix GetShuffleId RPC NPE for empty shuffle
+- [CELEBORN-1147] Added a dedicated API for RPC messages which also accepts an RpcResponseCallback instance
+- [MINOR] Update log level of ChunkFetchSuccess failed for `FetchHandler#handleChunkFetchRequest` from error to warn
+- [CELEBORN-1127] Add JVM classloader metrics
+- [CELEBORN-1122] Metrics supports json format
+- [CELEBORN-1052][FOLLOWUP] Introduce dynamic ConfigService at SystemLevel and TenantLevel
+- [CELEBORN-1145] Separate clientPushBufferMaxSize from CelebornInputStreamImpl
+- [CELEBORN-1131] Add Client/Server bootstrap framework to transport layer
+- [CELEBORN-1142] clear shuffleIdCache in shutdown method of ShuffleClientImpl
+- [CELEBORN-1081][FOLLOWUP] Remove UNKNOWN_DISK and allocate all slots to disk
+- [CELEBORN-1125][FOLLOWUP] Add failureaccess shade
+- [CELEBORN-1140] Use try-with-resources to avoid FSDataInputStream not being closed
+- [CELEBORN-1135] Added tests for the RpcEnv and related classes
+- [CELEBORN-856] Add mapreduce integration test
+- [CELEBORN-1123] Support fallback to non-columnar shuffle for schema that cannot be obtained from shuffle dependency
+- [CELEBORN-1134] Celeborn Flink client should validate whether execution.batch-shuffle-mode is ALL_EXCHANGES_BLOCKING
+- [CELEBORN-1106] Ensure data is written into flush buffer before sending message to client
+- [CELEBORN-1110][FOLLOWUP] Support celeborn.worker.storage.disk.reserve.ratio to configure worker reserved ratio for each disk
+- [CELEBORN-1108][FOLLOWUP] Use rat plugin check Flink 1.18
+- [CELEBORN-1048][FOLLOWUP] MR module compile
+- [CELEBORN-1108] Rat plugin check for more modules
+- [CELEBORN-247][FOLLOWUP] Add metrics for each user's quota usage of Celeborn Worker
+- [CELEBORN-1095] Support configuration of fastest available XXHashFactory instance for checksum of Lz4Decompressor
+- [CELEBORN-1087] Remove SimpleStateMachineStorageUtil in master module
+- Revert "[CELEBORN-255] Add counter of outstandingFetches, outstanding…
```

Comment:
such revert should not be present in release notes ...

let's remove it and the reverted one

Reply:
removed